### PR TITLE
Feature/migrate to v16

### PR DESCRIPTION
**Pull request by GagaMen**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit messages follow our guidelines
-   [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [x] Other... Please describe:

## What is the current behavior?

Angular version v15

## What is the new behavior?

Angular version v16

### CODE REVIEW

File: src/app/guards/configuration.guard.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { Configuration } from '../models/configuration';
 @Injectable({
     providedIn: 'root',
 })
-export class ConfigurationGuard implements CanActivate {
```
Comment (TobiDimmel) [inline src/app/guards/configuration.guard.ts:11]: This isn't a proper migration to the new style of defining guards.
I would prefer to migrate to the new style immediately.

See the links for the new style:
* https://angular.io/guide/deprecations#router-class-and-injectiontoken-guards-and-resolvers
* https://angular.io/guide/router#preventing-unauthorized-access

You can also use `ng g` to see how it is done.

**Review by TobiDimmel — CHANGES_REQUESTED** [review]
see my comments

File: package.json [hunk 2]
```diff
@@ -1,99 +1,97 @@
...
-        "@typescript-eslint/eslint-plugin": "^5.43.0",
-        "@typescript-eslint/parser": "^5.43.0",
-        "chance": "^1.1.7",
-        "commitizen": "^4.2.5",
-        "conventional-changelog-cli": "^2.2.2",
-        "cz-conventional-changelog": "3.3.0",
-        "eslint": "^8.28.0",
-        "eslint-config-prettier": "^8.5.0",
-        "eslint-plugin-import": "^2.26.0",
-        "eslint-plugin-jsdoc": "^39.3.6",
```
[the remark this answers is not in the document]
Comment (GagaMen) [reply package.json:69]: We don't use these package in our eslint config.

File: src/app/guards/configuration.guard.ts (hunk 1 above)
Comment (GagaMen) [reply src/app/guards/configuration.guard.ts:11]: done

**Comment by TobiDimmel** [comment]
LGTM

**Pull request merged**